src/mongocxx/options/find.hpp

```diff
@@ -308,6 +330,7 @@ class MONGOCXX_API find {
    private:
     stdx::optional<bool> _allow_partial_results;
     stdx::optional<std::int32_t> _batch_size;
+    stdx::optional<bsoncxx::document::view_or_value> _collation;
```

Comment:
So, note that while the addition of the above *methods* wouldn't constitute an ABI break, adding this field would. We aren't promising ABI yet, but, I feel it is important to identify places were it would happen so we gain familiarity. I suspect that at some point we are going to want to give all options an "extension" parameter slot, which is a unique_ptr to some opaque internal extensions object. Or we may find that we need to pimpl all options fields? I'm not sure.